### PR TITLE
administration: preview for existing form

### LINKED ISSUE #835: vTHL3: Preview form
When creating or editing a form, there should be a possibility to preview what it looks like to the applicant. This could open a new window, modal or show beside the editor on large screens.

- [x] preview action while creating a new form #1110 
- ~preview action while editing an existing form~ no form editing is planned
- [x] preview for existing form #1113 
- [x] style adjustments: editor and preview side-by-side

The form does not need to be functioning but it should look like what it will look like in a real application.

Plan of attack:
- [x] push current form creation model as is into db (still keeping old save functionality)
- [x] make an API for returning that model
- [x] UI preview with that model
- leftover refactoring opportunities in #913

### PR DESCRIPTION
**Pull request by opqdonut**
plus some form api cleanup

for #835
related to #913

### CODE REVIEW

File: src/cljs/rems/administration/form.cljs [hunk 1]
```diff
@@ -37,33 +38,6 @@
...
-         (for [localization (:localizations field)]
-           [inline-info-field (str (text :t.administration/title)
-                                   " "
-                                   (str/upper-case (name (:langcode localization)))) (:title localization)])
-         [[inline-info-field (text :t.create-form/type) (text (keyword (str "t.create-form/type-" (:type field))))]
-          [inline-info-field (text :t.create-form/input-prompt) (get-localized-value field :inputprompt language)]
-          [inline-info-field (text :t.create-form/optional) [readonly-checkbox (:formitemoptional field)]]
-          [inline-info-field (text :t.create-form/maxlength) (:maxlength field)]])))
-
-(defn form-fields [fields language]
```
Comment (opqdonut) [inline src/cljs/rems/administration/form.cljs]: I felt like these didn't add any value on top of the preview so I removed them

File: src/cljs/rems/administration/form.cljs [hunk 2]
```diff
@@ -75,8 +49,8 @@
               [inline-info-field (text :t.administration/start) (localize-time (:start form))]
               [inline-info-field (text :t.administration/end) (localize-time (:end form))]
               [inline-info-field (text :t.administration/active) [readonly-checkbox (not (:expired form))]]]}]
-   [form-fields (:fields form) language]
-   [:div.col.commands [back-button]]])
+   [:div.col.commands [back-button]]
```
Comment (opqdonut) [inline src/cljs/rems/administration/form.cljs]: back is between the info and the preview. could also be at the top I guess. I don't want to have it below the preview, and I don't think a collapsible preview would be any better

**Comment by Macroz** [comment]
I think the preview should not be open all the time, but rather open into a tab or maybe just modal container when you want. Perhaps it's also possible to split the view horizontally and render both on a wider display to actually use the space well but I'd go with the modal first.

Also noticed errors with THL form.

File: src/clj/rems/api/forms.clj [hunk 3]
```diff
@@ -26,13 +26,13 @@
    :optional s/Bool
    :type (s/enum "attachment" "date" "description" "label" "multiselect" "option" "text" "texta")
    (s/optional-key :maxlength) (s/maybe (s/constrained s/Int not-neg?))
-   (s/optional-key :options) [{:key s/Str
+   (s/optional-key :options) [{:key s/Any ;; s/Str for multiselect, s/Bool for option
```
Comment (luontola) [inline src/clj/rems/api/forms.clj]: Smells fishy.

Comment (opqdonut) [reply src/clj/rems/api/forms.clj]: I removed the fishes

**Comment by opqdonut** [comment]
hmm okay it seems the bool keys used in the thl form are actually invalid. let me fix those instead of the schema

File: test/clj/rems/api/test_forms.clj [hunk 4]
```diff
@@ -61,32 +61,75 @@
...
+                             :maxlength 300
+                             :input-prompt localized}
+                            {:type "description"
+                             :title localized
+                             :optional false}
+                            {:type "option"
+                             :title localized
+                             :optional true
+                             :options [{:key "true" :label localized}
+                                       {:key "false" :label localized}]}
```
Comment (luontola) [inline test/clj/rems/api/test_forms.clj]: Option fields can have more than two options. The difference between option and multiselect is that option lets the user choose only one value, but multiselect lets the user choose many values. To make that explicit, this example could be:
```clj
:options [{:key "a" :label localized}
          {:key "b" :label localized}
          {:key "c" :label localized}]

**Pull request merged**